packages/backend/src/open_payments/wallet_address/service.ts

```diff
@@ -166,7 +166,7 @@ async function createWalletAddress(
 
     return await WalletAddress.query(deps.knex)
       .insertGraphAndFetch({
-        url: options.url,
+        url: options.url.toLowerCase(),
```

Comment:
let's add some tests for these :)

Reply:
@s100tist 
in the catch statement, you can check for `err instanceof UniqueViolationError` to catch those cases where you are trying to create a duplicate wallet address